compiler-rt/lib/sanitizer_common/sanitizer_mutex.cpp

```diff
@@ -212,8 +212,8 @@ struct InternalDeadlockDetector {
     return initialized > 0;
   }
 };
-
-static THREADLOCAL InternalDeadlockDetector deadlock_detector;
+__attribute__((tls_model("initial-exec"))) static THREADLOCAL
```

Comment:
This deserves a comment.

Reply:
Sounds good will add one. I'll see if here is the best place or if there is another global declared with this attribute that works better.